### PR TITLE
Replace select 2 with choices.js and kill jQuery

### LINKED ISSUE #1001: Replace `jquery` with plain javascript
Replace `jquery` code with plain javascript for custom javascript code
`bootstrap` doesn't required jquery any more
`select2` still requires jquery. maybe we could also replace this to drop jquery completely

### PR DESCRIPTION
**Pull request by weeman1337**
closes #1001

### CODE REVIEW

**Review by MyIgel — CHANGES_REQUESTED** [review]
Looks like only the first select of a page gets used? On the shifts page only the from date is selectable, the to is the old select. Same for the room select on the shifts creation page
image

On the messages recepient select the text is too long:
Screenshot_20221224_212418

At least the Engelsystem dark theme needs some contrast improvements:
Screenshot_20221224_212554

**Comment by ximex** [comment]
see: https://github.com/Choices-js/Choices#setup
> Note: If you pass a selector which targets multiple elements, the first matching element will be used. Versions prior to 8.x.x would return multiple Choices instances.

**Comment by weeman1337** [comment]
> Looks like only the first select of a page gets used? 

Done

> On the messages recepient select the text is too long:

Done

File: resources/assets/js/forms.js [hunk 1]
```diff
@@ -200,10 +200,15 @@ ready(() => {
 });
 
 ready(() => {
-  $('select').select2({
-    theme: 'bootstrap-5',
-    width: '100%',
-  });
+  document.querySelectorAll('select').forEach(
+    (element) =>
```
Comment (ximex) [inline resources/assets/js/forms.js:204]: i would put the `(element) =>` into the previous line

Comment (weeman1337) [reply resources/assets/js/forms.js:204]: Code style in JS/CSS is handled by [prettier](https://prettier.io/).

**Comment by weeman1337** [comment]
> At least the Engelsystem dark theme needs some contrast improvements:

Dark theme contract is bad anyway :stuck_out_tongue: 

image

**Comment by ximex** [comment]
this PR should get rebased on main to have this commit: https://github.com/engelsystem/engelsystem/commit/30f50dab6d758977b3a2e763a49f511b60986fd9
this is related to the today/yesterday/tomorrow buttons

**Comment by MyIgel** [comment]
@ximex That literally doesn't help in any way with any of the above problems so yes it could be rebased but that's literally useless.

**Comment by weeman1337** [comment]
> Directly using full white might be questionable but works at least.

select_after

tabs

I do only use the same colours for selected elements, like for instance tabs. If you want other colours here, please send me them and I will change them.

> More problematic is the point that the "Today" / "Yesterday" / "Tomorrow" buttons stopped working.

Done

> Also (at least in my setup with many shifts) there seems to be some weird caching issue or so when searching -> we should have a look at this too.

Can you explain the issue?

**Comment by MyIgel** [comment]
Looks good so far, the issue is that the select seems to only show the first four results when searching. When i search for 2022 i expect all 2022 dates to show up but only get the 1st January until the 4th January and not more.

**Comment by weeman1337** [comment]
> only show the first four results when searching. 

done

Design now provides a 50 options list for testing.

**Comment by weeman1337** [comment]
:ping_pong: @MyIgel can you re-test the search?

**Pull request merged**